packages/fast-usdc/src/utils/store.js

```diff
@@ -0,0 +1,114 @@
+import { Fail } from '@endo/errors';
+
+/**
+ * @import {Key} from '@endo/patterns';
+ */
+
+// TODO provide something like this in a more common place, perhaps as a BagStore
+/**
+ * Creates a bag (multi-set) API that wraps a MapStore where values are counts.
+ *
+ * @template {Key} K
+ * @param {MapStore<K, number>} mapStore
+ */
+export const asMultiset = mapStore =>
+  harden({
+    /**
+     * Add an item to the bag, incrementing its count.
+     *
+     * @param {K} item The item to add
+     * @param {number} [count] How many to add (defaults to 1)
+     */
+    add: (item, count = 1) => {
+      if (count <= 0) {
+        throw Fail`Cannot add a non-positive count ${count} to bag`;
+      }
```

Comment:
Probably should check that `count` is a Nat ?

Reply:
I'm using number and Nat is bigint

Reply:
Sure, but we should still enforce that this in an integer, not any float.

Reply:
Actually it's not just floats, but any NaN or non number values that will cause unexpected value for current count.